### PR TITLE
gh-99204: Calculate base_executable by alternate names in POSIX venvs

### PR DESCRIPTION
**Pull request by vfazio**
Check to see if `base_executable` exists. If it does not, attempt to use known alternative names of the python binary to find an executable in the path specified by `home`.

If no alternative is found, previous behavior is preserved.

* Issue: gh-99204

### CODE REVIEW

**Review by FFY00 — APPROVED** [review]
I think this is a reasonable fix, but let's see if others agree.

File: Modules/getpath.py
Comment (zooba) [inline Modules/getpath.py]: Leading underscores on variables isn't a convention that we use in this file.

**Comment by zooba** [comment]
Conceptually it seems fine. Not sure whether we should mark this a "QUIRK" and aim to resolve it in some other way? I'll leave that to the POSIX people to decide

File: Modules/getpath.py
Comment (vfazio) [reply Modules/getpath.py]: addressed

**Comment by zooba** [comment]
Please don't force push changes when doing PRs. We'll squash merge at the end, but for now I have to re-review from the start because the diff is broken.

I don't have time for that right now, but if someone else signs off and the variable names are indeed fixed, they can merge.

**Comment by vfazio** [comment]
would there be any concerns about me also making a PR targeting 3.11 ?

**Pull request merged**
